test/test_jit.py

```diff
@@ -4252,6 +4252,30 @@ def bar(x):
 
         self.assertEqual(8, bar(torch.ones(1, 1)))
 
+    def test_tracing_slicing(self):
+        @torch.jit.trace(torch.zeros(10))
+        def foo_trace(x):
+            return x[-1:-3]
+
+        @torch.jit.script
+        def foo_script(x):
+            return x[-1:-3]
+
+        a = torch.arange(0, 8)
+        self.assertEqual(foo_script(a), foo_trace(a))
```

Comment:
Actually the biggest problem with wrapping negative indices is that it specializes them to a particular size. Can you have a test which runs this function on two differently sized inputs and gets two differently sized outputs as a result?

Reply:
The traced tensor and tensor that I test it on are a different size but I'll add another example.